grades/models.py

```diff
@@ -223,7 +225,8 @@ def for_user_course(cls, user, course):
         """
         Returns a queryset of the exam result for an user in a course of a program
         """
-        return cls.objects.filter(user=user, course=course)
+        grades_available_runs = ExamRun.get_all_with_available_grades(course)
+        return cls.objects.filter(user=user, course=course, exam_run__in=grades_available_runs)
```

Comment:
The two separate queries here (in `ProctoredExamGrade`) and in `ExamRun` cause the resulting DB query to have a subquery, ie: `select * from grades_proctoredexamgrade where course_id in (select course_id from exam_run ... )`. Not the end of the world obviously, but it kinda seems like that logic is only relevant in this model (this is the only client of `ExamRun.get_all_with_available_grades`, and I would argue that it's likely to stay that way). If you perform the query for `date_grades_available__lte=now` in this method, we avoid the nested query and we keep the logic more proximal. What do you think?